content/docs/platform/subscription/headless-hooks.mdx

```diff
@@ -0,0 +1,157 @@
+---
+pageTitle: 'Headless hooks'
+title: 'Headless hooks'
+description: 'Build a fully custom subscription interface using hooks.'
+icon: 'Code'
+---
+
+Novu `@novu/react` package provides Subscription hooks that let you build your own custom subscription experiences from scratch.
+
+These hooks allow you to list, create, update, and delete subscriptions while leveraging Novu's state management.
+
+<Callout type="info"> For the full list hooks, props, and methods, refer to the [@novu/react SDK documentation](/sdks/react). Not using React? Check out the [@novu/js SDK documentation](/sdks/javascript). For a complete reference on all available Subscription endpoints, refer to the [API reference documetation](/api-reference/overview).</Callout>
+
+## Get all subscriptions
+
+The `useSubscriptions` hook retrieves all subscriptions associated with a specific topic for the current user. This is useful for rendering a list of "My Subscriptions" where users can see everything they're opted into.
+
+```tsx
+import { useSubscriptions } from '@novu/react';
+
+function SubscriptionList() {
+  const { subscriptions, isLoading } = useSubscriptions({ 
+    topicKey: 'product-updates' 
+  });
+
+  if (isLoading) return <div>Loading...</div>;
+
+  return (
+    <ul>
+      {subscriptions.map((subscription) => (
+        <li key={subscription.id}>
+           {/* If name is undefined, fall back to the identifier */}
+           <span>{subscription.name || subscription.identifier}</span>
+        </li>
+      ))}
+    </ul>
+  );
+}
+```
+
+## Get a single subscription
+
+If you need to manage a specific subscription instance, use the `useSubscription` hook. It fetches the details and preference states for a single subscription.
+
+```tsx
+import { useSubscription } from '@novu/react';
+
+function ProjectSettings() {
+  const { subscription, isLoading } = useSubscription({ 
+    topicKey: 'project-123',
+    identifier: 'user-project-alert'
+  });
+
+  if (isLoading) return <div>Loading...</div>;
+
+  if (!subscription) return <div>You are not subscribed to this project.</div>;
+
+  return (
+    <div>
+      <h3>Subscription: {subscription.identifier}</h3>
+      <pre>{JSON.stringify(subscription.preferences, null, 2)}</pre>
+    </div>
+  );
+}
+```
+
+## Create a subscription
+
+To allow users to opt-in to a topic, use the `useCreateSubscription` hook. This is often used on "Follow" or "Subscribe" buttons.
+
+```tsx
+import { useCreateSubscription } from '@novu/react';
+
+function SubscribeButton() {
+  const { create, isCreating } = useCreateSubscription();
+
+  const handleSubscribe = async () => {
+    try {
+      await create({ 
+        topicKey: 'project-123', 
+        identifier: 'user-project-alert',
+        // Optional: Set initial preferences
+        preferences: [] 
+      });
```

Comment:
There is no need for a `try/catch` block because the `create` never throws and has the same interface as all the `@novu/js` package actions, I mean it returns `const { data, error } = await create({...});`